hydromt/models/components/mesh.py

```diff
@@ -10,34 +10,389 @@
 import xarray as xr
 import xugrid as xu
 from pyproj import CRS
+from shapely.geometry import box
 
-from hydromt import workflows
 from hydromt.gis.raster import GEO_MAP_COORD
+from hydromt.io.readers import read_nc
 from hydromt.models.api import Model
+from hydromt.models.components.base import ModelComponent
+from hydromt.workflows.mesh import (
+    create_mesh2d,
+    mesh2d_from_raster_reclass,
+    mesh2d_from_rasterdataset,
+    rename_mesh,
+)
 
-__all__ = ["MeshModel"]
-logger = logging.getLogger(__name__)
+logger = getLogger(__name__)
 
 
-class MeshMixin(object):
-    # placeholders
-    # We cannot initialize an empty xu.UgridDataArray
-    _API = {
-        "mesh": Union[xu.UgridDataArray, xu.UgridDataset],
-    }
+class MeshComponent(ModelComponent):
+    """ModelComponent class for mesh components."""
 
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self._mesh = None
+    def __init__(self, model: Model):
+        super().__init__(model=model)
+        self._data = None
 
-    ## general setup methods
-    def setup_mesh2d_from_rasterdataset(
+    def set(
+        self,
+        data: Union[xu.UgridDataArray, xu.UgridDataset],
+        name: Optional[str] = None,
+        grid_name: Optional[str] = None,
+        overwrite_grid: bool = False,
+    ) -> None:
```

Comment:
```suggestion
    def set(
        self,
        data: Union[xu.UgridDataArray, xu.UgridDataset],
        *,
        name: Optional[str] = None,
        grid_name: Optional[str] = None,
        overwrite_grid: bool = False,
    ) -> None:
```

I would make all naming mandatory